src/Adapter/MSTest.TestAdapter/Execution/ExceptionHelper.cs

```diff
@@ -145,10 +145,20 @@ internal static string GetFormattedExceptionMessage(this Exception ex)
 
         StringBuilder result = new();
         bool first = true;
+        HashSet<Exception> seenExceptions = new();
+
         for (Exception? curException = ex;
              curException != null;
              curException = curException.InnerException)
         {
+            if (seenExceptions.Contains(curException))
+            {
+                result.Append(" ---> [Cyclic Exception Reference]");
+                break;
+            }
+            
+            seenExceptions.Add(curException);
```

Comment:
Could be reduced to:

```suggestion
            if (!seenExceptions.Add(curException))
            {
                result.Append(" ---> [Cyclic Exception Reference]");
                break;
            }
```